sklearn/ensemble/_hist_gradient_boosting/tests/test_loss.py

```diff
@@ -70,20 +72,26 @@ def test_derivatives(loss, x0, y_true):
     x0 = np.array([x0], dtype=Y_DTYPE).reshape(1, 1)
     get_gradients, get_hessians = get_derivatives_helper(loss)
 
-    def func(x):
-        return loss.pointwise_loss(y_true, x)
+    def func(x: float) -> float:
+        if isinstance(loss, _LOSSES['binary_crossentropy']):
+            # Subtract a constant term such that the binary cross entropy
+            # has its minimum at zero. This only works if 0 < y_true < 1.
+            return (loss.pointwise_loss(y_true, x)
+                    - loss.pointwise_loss(y_true, logit(y_true))).item(0)
+        else:
+            return loss.pointwise_loss(y_true, x).item(0)
 
-    def fprime(x):
-        return get_gradients(y_true, x)
+    def fprime(x: float) -> float:
+        return get_gradients(y_true, x).item(0)
 
-    def fprime2(x):
-        return get_hessians(y_true, x)
+    def fprime2(x: float) -> float:
+        return get_hessians(y_true, x).item(0)
```

Comment:
instead of dealing with indices here (and below), maybe we should instead do this just before the assertsions:

```py
    y_true = y_true.ravel()
    optimum = optimum.ravel()
```

Reply:
See comment above. AFAIU, the `newton` root finder prefers scalars.
Maybe, I can't see the forest for the trees: `np.array([1.]).ravel()` still returns a ndarray.

Reply:
newton works fine with arrays (as long as they have one element like here).

> np.array([1.]).ravel() still returns a ndarray.

Yes, but they would all have the same shape, and it would make `assert_allclose` happy.

The changes you had to do with the use of `item()` and stuff come from the fact that you're now using `assert_allclose` which cannot  compare arrays with different dimensions. It's not related to newton.

Reply:
I see. 👍

Reply:
lol yeah I had to do a few back and forth with master myself to really understand what was going on